orchestrator/expiration.go

```diff
@@ -0,0 +1,64 @@
+package orchestrator
+
+import (
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"sort"
+	"time"
+)
+
+const (
+	DefaultExpirationDuration = 10 * time.Second
+)
+
+type expirationRecord struct {
+	offset kafka.TopicPartition
+	expirationTime time.Time
+}
+
+type ExpirationMonitor struct {
+	records []*expirationRecord // kafka message offsets ordered by expiration time
+}
+
+func NewExpirationMonitor() *ExpirationMonitor {
+	return &ExpirationMonitor{
+		make([]*expirationRecord, 0),
+	}
+}
+
+func (e *ExpirationMonitor) Add(offset kafka.TopicPartition) {
+	now := time.Now()
+	expirationTime := now.Add(DefaultExpirationDuration)
```

Comment:
Perhaps you make the DefaultExpirationDuration a property of ExpirationMonitor?

Reply:
Yep, I was planning to do that as part of the work that will prepare it for deployment it in k8s.

Reply:
Fixed